### DISCUSSION TITLE
Issues with extracting tables with graphics

### DISCUSSION THREAD

**Discussion by florpesoa** [jsvine/pdfplumber#1046] [Ask for help with specific PDFs]
Hello! I'm new to programming, and I'm using the library. It's been very helpful, but I'm experiencing something. The tables I'm extracting are not very simple; they have some graphics and some things that it reads with extract_text, but not with extract_table, and I don't understand why. When I convert it to an image to see what it reads, it highlights it in red and everything. Any idea why this could be? Thanks!

**Comment by florpesoa** [comment]
def process_plumber3(pdf_path):
  with pdfplumber.open(pdf_path) as pdf:
  page = pdf.pages[13-1]
  header_area = (0, 0, page.width, 200) 
     
  header_text = page.within_bbox(header_area, strict=False).extract_table(table_settings={"horizontal_strategy": "lines", "text_y_tolerance": 15 })
  df=process_plumber3(pdf_path)
  df=pd.DataFrame(df)


I used python 3.8
<img width="297" alt="copy" src="https://github.com/jsvine/pdfplumber/assets/111348473/a8db1ab6-04e1-4e30-b1ab-d48c9b04ca6e">
Thanks!

**Comment by florpesoa** [comment]
<img width="383" alt="output" src="https://github.com/jsvine/pdfplumber/assets/111348473/b68f85c0-c0b9-4beb-9798-1d54133f44e0">

**Reply by jsvine** [reply]
Hi @florpesoa, this looks like you might have gotten the code to work? If not, can you describe the difference between what you're getting and what you're hoping to get? Also, are you able to provide the PDF?

**Comment by florpesoa** [comment]
Hi!!
The difference in the output is that it doesn't fully read what's in the PDF, for example: where the output begins is where the blue color is, but it doesn't read everything above that. However, when I want to see what it does read from the PDF, it does highlight all the rest of the text in red.
I don't quite understand why not, but I was able to solve it with the following code that I saw in the discussions. Thank you very much for the quick response :
If I change the values in row.bbox, it still doesn't work.

Apologies, I can't share the PDF due to confidentiality reasons. :(
```
                    page = doc.pages[page_number-1]
                    explicit_horizontal_lines = set().union(*((row.bbox[0], row.bbox[2]) for row in (page.find_tables())[0].rows))
                    explicit_horizontal_lines.add(max(page.chars, key=itemgetter("bottom"))["bottom"])
                    table = page.extract_table(dict(
                    explicit_horizontal_lines=explicit_horizontal_lines, 
                    horizontal_strategy="text", text_x_tolerance= 1, text_y_tolerance=1 
                        ))

**Reply by jsvine** [reply]
Thank you. Unfortunately, this will be very difficult to answer without access to the PDF. But one step you could take is to examine the results of `page.to_image().debug_tablefinder({ ... })`.
